src/core_plugins/kibana/public/dashboard/dashboard.html

```diff
@@ -28,7 +28,6 @@
           <div class="kuiLocalSearch">
             <input
               parse-query
-              input-focus
```

Comment:
If auto-focus is a critical usability improvement, then I think we should keep it and figure out other ways to address the accessibility concerns, e.g. providing context to screen readers with aria attributes.

FYI @snide is working on some designs this week that incorporate both these changes and the ones in https://github.com/elastic/kibana/pull/10976.